proxy/IPAllow.cc

```diff
@@ -259,8 +259,9 @@ IpAllow::BuildTable()
   std::error_code ec;
   std::string content{ts::file::load(config_file, ec)};
   if (ec.value() == 0) {
-    // If it's a .yaml or the root tag is present, treat as YAML.
-    if (TextView{config_file.view()}.take_suffix_at('.') == "yaml" || std::string::npos != content.find(YAML_TAG_ROOT)) {
+    // If it's a .yaml or the tokens "apply" *and* "ip_addrs" are present, treat as YAML.
+    if (TextView{config_file.view()}.take_suffix_at('.') == "yaml" ||
+        (std::string::npos != content.find(YAML_TAG_IP_ADDRS) && std::string::npos != content.find(YAML_TAG_APPLY))) {
       this->YAMLBuildTable(content);
     } else {
       this->ATSBuildTable(content);
```

Comment:
Why are we supporting 2 types of configuration files.  We should get rid of the old configuration format.

Reply:
Because that was was agreed upon at the time the YAML support for IPAllow PR was created, back in August of 2018. When you make PRs hang out for a year, you get archaic results.

Reply:
+1 to @SolidWallOfCode . We need to support both configs for at least one version in order to allow people to transition from one to the other. Otherwise, everybody has to cut over all at once, which is a fair bit harder.